src/record-error-archival/src/recordErrorArchival.ts

```diff
@@ -0,0 +1,63 @@
+import type { ApiClient } from "shared-types"
+import { AuditLogEvent, isError, isSuccess } from "shared-types"
+import type { ArchivedErrorRecord } from "./DatabaseClient"
+import type DatabaseClient from "./DatabaseClient"
+import groupBy from "lodash.groupby"
+import { logger } from "shared"
+
+export const recordErrorArchival = async (db: DatabaseClient, api: ApiClient) => {
+  await db.connect()
+  const errorRecords = await db.fetchUnloggedArchivedErrors()
+
+  const results: { success: boolean; reason?: string; errorRecord: ArchivedErrorRecord }[] = []
+
+  for (const errorRecord of errorRecords) {
+    logger.debug({ message: "Would audit log the archival of an error", record: errorRecord })
+
+    const auditLogEvent = new AuditLogEvent({
+      eventSource: errorRecord.archivedBy,
+      category: "information",
+      eventType: "Error archival",
+      timestamp: errorRecord.archivedAt
+    })
+    auditLogEvent.addAttribute("Error ID", errorRecord.errorId)
+
+    const response = await api.createEvent(errorRecord.messageId, auditLogEvent)
```

Comment:
Before creating the event, we need to check to make sure that we haven't already created an event for this error. We probably can do that by checking `Error ID` attribute value.